File: CleverFerretV2/core/data/src/main/java/com/cleverferret/v2/core/data/contracts/UnifiedMediaItemV1.java
package com.cleverferret.v2.core.data.contracts;

import com.cleverferret.v2.core.common.VersionedContract;

public record UnifiedMediaItemV1(
        String itemId,
        String libraryId,
        String title,
        MediaTypeV1 mediaType,
        String filePath
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/core/data/src/main/java/com/cleverferret/v2/core/data/contracts/UnifiedMediaItemV1.kt
package com.cleverferret.v2.core.data.contracts

import com.cleverferret.v2.core.common.VersionedContract

data class UnifiedMediaItemV1(
    val itemId: String,
    val libraryId: String,
    val title: String,
    val mediaType: MediaTypeV1,
    val filePath: String
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
